app/routers/game_release_dates_service.py

```diff
@@ -0,0 +1,181 @@
+import datetime
+from http import HTTPStatus
+
+from fastapi import APIRouter, Depends, Request
+from fastapi.responses import HTMLResponse
+from loguru import logger
+from sqlalchemy.orm import Session
+from starlette.responses import RedirectResponse
+from starlette.status import HTTP_302_FOUND
+from starlette.templating import _TemplateResponse
+
+import requests
+from app.database.models import UserSettings
+from app.dependencies import get_db, templates
+from app.internal.security.dependencies import current_user
+from app.internal.security.schema import CurrentUser
+from app.internal.utils import create_model
+
+# from typing import Dict, List, Union
+
+
+router = APIRouter(
+    prefix="/game-releases",
+    tags=["game-releases"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+def is_user_signed_up_for_game_releases(
+    session: Session,
+    current_user_id: int,
+):
+    is_signed_up = (
+        session.query(UserSettings)
+        .filter(UserSettings.user_id == current_user_id)
+        .filter(UserSettings.video_game_releases.is_(True))
+        .first()
+    )
+
+    if is_signed_up:
+        return True
+    return False
+
+
+# @router.post("/add-games-to-calendar")
+# async def add_games_to_calendar(request: Request, session=Depends(get_db)):
+#     current_user = get_current_user(session)
+#     data = await request.form()
+
+#     print(data)
+#     for game in data:
+#         print(game)
+#         game_data = requests.get(
+#             f"https://api.rawg.io/api/games/{game[0]}",
+#         ).json()
+
+#         create_event(
+#             session,
+#             title=game["name"],
+#             start=game["release_date"],
+#             end=game["release_date"],
+#             owner_id=current_user.id,
+#         )
+#         print(game_data)
+
+
+@router.post("/get_releases_by_dates")
+async def fetch_released_games(request: Request, session=Depends(get_db)):
+    data = await request.form()
+
+    from_date = data["from_date"]
+    to_date = data["to_date"]
+
+    template = templates.get_template(
+        "partials/calendar/feature_settings/games_list.html",
+    )
+    games = get_games_data(from_date, to_date)
+    content = template.render(games=games)
+    return HTMLResponse(content=content, status_code=HTTPStatus.OK)
+
+
+@router.get("/get_game_releases_next_month")
+def get_game_releases_month(request: Request):
+    today = datetime.datetime.today()
+    delta = datetime.timedelta(days=30)
+    today_str = today.strftime("%Y-%m-%d")
+    in_month_str = (today + delta).strftime("%Y-%m-%d")
+
+    games = get_games_data(today_str, in_month_str)
+
+    return games
+
+
+# @router.post
+# TODO
+# add game to calendar
+# @router.post
+# remove game from calendar
```

Comment:
feature should be shipped with these.

Reply:
Currently I decided not to add the releases to DB (in case the game release updates it would be worse than not knowing the date at all) and only update real time from the API.